lib/io/can.hpp

```diff
@@ -23,10 +27,18 @@ struct CanFrame {
 
 enum class CanResult { kFailure, kSuccess };
 
+class ICanProcessor {
+ public:
+  virtual void processMessage(const io::CanFrame frame);
+};
+
 class ICan {
+ public:
   virtual CanResult initialise(const std::string &can_network_interface);
   virtual CanResult send(const CanFrame &message);
   virtual std::optional<CanFrame> receive();
+  virtual CanResult addCanProcessor(uint16_t ID, ICanProcessor processor);
```

Comment:
`const`

`ID` is upper case

be careful with passing by value vs passing by reference here; copying this over could be a problem? I don't know. The safest thing is to require a `std::shared_ptr`.